apps/cowswap-frontend/src/modules/tokensList/containers/TokenSearchResults/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { SelectTokenContext } from '../../types'
+
+const searchResultsLimit = 10
+
+export interface TokenSearchResultsProps extends SelectTokenContext {
+  isEnterPressed: boolean
+  searchInput: string
+}
+
+export function TokenSearchResults({
+  searchInput,
+  balances,
+  selectedToken,
+  onSelectToken,
+  unsupportedTokens,
+  permitCompatibleTokens,
+  isEnterPressed,
+}: TokenSearchResultsProps) {
+  const { inactiveListsResult, blockchainResult, activeListsResult, externalApiResult, isLoading } =
+    useSearchToken(searchInput)
+
+  const addTokenImportCallback = useAddTokenImportCallback()
+
+  const networkName = useNetworkName()
+
+  const searchCount = useMemo(() => {
+    return [
+      activeListsResult?.length,
+      inactiveListsResult?.length,
+      blockchainResult?.length,
+      externalApiResult?.length,
+    ].reduce<number>((acc, cur) => acc + (cur ?? 0), 0)
+  }, [activeListsResult?.length, inactiveListsResult?.length, blockchainResult?.length, externalApiResult?.length])
+
+  const isTokenNotFound = useMemo(() => {
+    if (isLoading) return false
+
+    return searchCount === 0
+  }, [isLoading, searchCount])
+
+  // On press Enter, select first token if only one token is found or it's fully matches to the search input
+  useEffect(() => {
+    if (!isEnterPressed || !searchInput || !activeListsResult) return
+
+    const matchedToken = activeListsResult.find((token) => doesTokenMatchSymbolOrAddress(token, searchInput))
+
+    if (activeListsResult.length === 1 || matchedToken) {
+      onSelectToken(matchedToken || activeListsResult[0])
+    }
+  }, [isEnterPressed, searchInput, activeListsResult, onSelectToken])
+
+  if (isTokenNotFound)
+    return <styledEl.TokenNotFound>No tokens found for this name in {networkName}</styledEl.TokenNotFound>
+
+  return (
+    <CommonListContainer id="currency-list">
+      {/*Tokens from active lists*/}
+      {activeListsResult &&
+        activeListsResult.slice(0, searchResultsLimit).map((token) => {
+          const addressLowerCase = token.address.toLowerCase()
+
+          return (
+            <TokenListItem
+              key={token.address}
+              isUnsupported={!!unsupportedTokens[addressLowerCase]}
+              isPermitCompatible={permitCompatibleTokens[addressLowerCase]}
+              selectedToken={selectedToken}
+              token={token}
+              balance={balances ? balances[token.address]?.value : undefined}
+              onSelectToken={onSelectToken}
+            />
+          )
+        })}
+
+      {/*Tokens from blockchain*/}
+      {blockchainResult?.length ? (
+        <styledEl.ImportTokenWrapper id="currency-import">
+          {blockchainResult.slice(0, searchResultsLimit).map((token) => {
+            return <ImportTokenItem key={token.address} token={token} importToken={addTokenImportCallback} />
+          })}
+        </styledEl.ImportTokenWrapper>
+      ) : null}
```

Comment:
This is new right?
I think it should have lower priority compared to inactive lists results.
Or, why do you think it should be here?

Reply:
The blockchain result has higher priority since it only searches by address, so, I think it should be first